syntax = "proto3";

package user_service;

service UserService {
  rpc RegisterUser (RegisterUserRequest) returns (RegisterUserResponse);
  rpc AuthenticateUser (AuthenticateUserRequest) returns (AuthenticateUserResponse);
  rpc GetUserProfile (GetUserProfileRequest) returns (GetUserProfileResponse);
}

message RegisterUserRequest {
  string email = 1;
  string username = 2;
  string password = 3;
}
message RegisterUserResponse {
  string user_id = 1;
  string status = 2;
}

message AuthenticateUserRequest {
  string email = 1;
  string password = 2;
}
message AuthenticateUserResponse {
  string user_id = 1;
  string token = 2;
  string status = 3;
}

message GetUserProfileRequest {
  string user_id = 1;
}
message GetUserProfileResponse {
  string user_id = 1;
  string email = 2;
  string username = 3;
  string created_at = 4;
}
